libr/bin/p/bin_mach0.c

```diff
@@ -503,6 +507,179 @@ static RBinInfo *info(RBinFile *bf) {
 	return ret;
 }
 
+static bool _patch_reloc(struct MACH0_(obj_t) *bin, RIOBind *iob, struct reloc_t * reloc, ut64 symbol_at) {
+	ut64 pc = reloc->addr;
+	ut64 ins_len = 0;
+
+	switch (bin->hdr.cputype) {
+	case CPU_TYPE_X86_64: {
+		switch (reloc->type) {
```

Comment:
This looks quite bad, could you place this `switch` in a separate function `handle_patch_reloc_x86_64(struct reloc_t *reloc, ut64 *pc, ut64 *ins_len);`? Same for the aarch64/arm case. It will be easily inlined by the compiler.

Reply:
is this for avoiding the nested switch?

Reply:
Yes. We are already full of nested switches everywhere and they become every day worse

Reply:
let me see if i can refactor this in a way which isn't worse than that 😅

Reply:
If you prefer nested switches I'm "ok" with that. Hopefully they will remain quite simple and short :)